### PR TITLE
fix: add support for windows builds

### LINKED ISSUE #14: Support windows builds
The current method of connecting to the docker socket, via the `unix://` path, does not work on windows systems. Connecting in an alternative way would enable support for windows.

### PR DESCRIPTION
**Pull request by tylerslaton**
## Summary
Adding some conditional compilation that enables builds to finish in windows environments. Prior to this PR, `xdg` was not allowing our builds to be cross-platform compatible. This conditionally compiles that import and sets a logical default for supporting Docker (not Podman, yet) in a windows environment.

Tangentially, this also updates the error messaging for not being able to connect to sockets to be clearer.

## Todo
1. Figure out how to setup an action to test that windows builds are working
2. Update documentation to explain how to use `dcp` on Windows

Closes #14

### CODE REVIEW

File: src/runtime.rs [hunk 1]
```diff
@@ -1,15 +1,28 @@
-use anyhow::{anyhow, Result};
 use docker_api::Docker;
+
+// Imports not used by windows environments
+#[cfg(not(target_os = "windows"))]
 use podman_api::Podman;
+#[cfg(not(target_os = "windows"))]
+use anyhow::Result;
+#[cfg(not(target_os = "windows"))]
+use anyhow::anyhow;
```
Comment (exdx) [inline src/runtime.rs:9]: Do we need to gate these imports? I imagine we can use anyhow regardless of the target OS?

**Review by exdx — APPROVED** [review]
Looks good. I think we can spin up windows VMs via GitHub Actions and test this out more, hopefully. Also we can add to the docs about windows support. These can both be follow-ups though.

File: src/runtime.rs (hunk 1 above)
Comment (tylerslaton) [reply src/runtime.rs]: Yep, so as the comment says, these are imports that windows environments don't use. We can totally import these regardless of the environment but if you're working on `dcp` in a windows environment, or build it in a windows environment, then you'll get an error here for unused imports. Its just a warning though, so nothing breaking.

Comment (exdx) [reply src/runtime.rs]: Ohh interesting. I would prefer no warnings, so let's keep this as-is.

**Pull request merged**